File: client-app/src/app/pages/department-page/model/entity/department.ts
export class Department {
    public id: number;
    public name: string;
    public createdAt: string;
    public departments: Department[];

    public forAddNew: boolean;
    public parentId: number | null;

    constructor() {
        this.forAddNew = false;
        this.parentId = -1;

        this.id = -1;
        this.name = '';
        this.createdAt = '';
        this.departments = [];
    }
}

File: WebService.Infrastructure/Entity/Department.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Xml.Serialization;

namespace WebService.Infrastructure.Entity
{
    [Table("dapartment")]
    [Serializable]
    public class Department
    {
        [Column("id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int Id { get; set; }

        [Column("name")]
        public string Name { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; }

        [Column("department_id")]
        public int? DepartmentId { get; set; }

        public List<Worker> Worker { get; set; }

        [JsonIgnore]
        [XmlIgnore]
        public Department DepartmentMain { get; set; }
        public List<Department> Departments { get; set; }

    }
}
